extensions/iceberg/src/main/java/io/deephaven/iceberg/util/BuildCatalogOptions.java

```diff
@@ -30,11 +31,38 @@ public String name() {
     }
 
     /**
-     * The catalog properties. Must contain {@value CatalogUtil#ICEBERG_CATALOG_TYPE} or
+     * The catalog properties provided by the user. Must contain {@value CatalogUtil#ICEBERG_CATALOG_TYPE} or
      * {@value CatalogProperties#CATALOG_IMPL}.
+     * <p>
+     * For building the catalog, use {@link #updatedProperties()} instead.
      */
     public abstract Map<String, String> properties();
 
+    /**
+     * Whether Deephaven should not inject any additional properties (like for disabling CRT, setting a custom client
+     * credentials provider, etc.) into the properties map, if not already set by the user. This is enabled by default.
+     * <p>
+     * Disabling it would require users to set all the properties themselves.
```

Comment:
Let's not call out any specific examples here, as it will likely change over time. I would say something generic along the lines of "this is used to work around any upstream issues and to provide settings that are valuable and may be necessary for Deephaven's usage of Iceberg".